cdap-app-templates/cdap-etl/cdap-data-pipeline-base/src/main/java/io/cdap/cdap/datapipeline/service/ConnectionUtils.java

```diff
@@ -121,4 +124,24 @@ public static ConnectorContext getConnectorContext(ServicePluginConfigurer plugi
     SimpleFailureCollector failureCollector = new SimpleFailureCollector();
     return new DefaultConnectorContext(failureCollector, pluginConfigurer);
   }
+
+  /**
+   * Returns {@link ConnectorSpec}
+   *
+   * @return filtered {@link ConnectorSpec} on the basis of plugin name and type
+   */
+  public static ConnectorSpec filterSpecWithPluginNameAndType(ConnectorSpec spec, @Nullable String pluginName,
+                                                              @Nullable String pluginType) {
+    if (Strings.isNullOrEmpty(pluginName) || Strings.isNullOrEmpty(pluginType)) {
```

Comment:
I think this can be `&&`, we should be able to filter by both name and type, or just with name or type